melodies_monet/plots/aircraftplots.py

```diff
@@ -0,0 +1,842 @@
+    if label is not None:
+        plot_dict['label'] = label
+    if vmin is not None and vmax is not None:
+        plot_dict['ylim'] = [vmin, vmax]
+    # Scale the fontsize for the x and y labels by the text_kwargs
+    plot_dict['fontsize'] = text_kwargs['fontsize'] * 0.8
+
+      
+                         
+    # Then, if no plot has been created yet, create a plot and plot the obs
+    if ax is None: 
+        # First define the colors for the observations
+        obs_dict = dict(color='k', linestyle='-', marker='*', linewidth=1.2, markersize=6.)
+        if plot_dict is not None:
+            # Whatever is not defined in the yaml file is filled in with the obs_dict here
+            plot_kwargs = {**obs_dict, **plot_dict} 
+        else:
+            plot_kwargs = obs_dict
+        # Create the figure
+        if fig_dict is not None:
+            f, ax = plt.subplots(**fig_dict)    
+        else: 
+            f, ax = plt.subplots(figsize=(10, 6))
+       
+        # Bin the altitude variable and calculate median and interquartiles
+        altitude_bins = pd.cut(df[altitude_variable], bins=bins)
+        # Calculate the midpoints of the altitude bins
+        bin_midpoints = altitude_bins.apply(lambda x: x.mid)
+        # Convert bin_midpoints to a column in the DataFrame
+        df['bin_midpoints'] = bin_midpoints
+        median = df.groupby(altitude_bins)[column].median()
+        q1 = df.groupby(altitude_bins)[column].quantile(0.25)
+        q3 = df.groupby(altitude_bins)[column].quantile(0.75)
+        # Convert bin_midpoints to a numerical data type
+        df['bin_midpoints'] = df['bin_midpoints'].astype(float)
+
+        p5 = df.groupby(altitude_bins)[column].quantile(0.05)
+        p10 = df.groupby(altitude_bins)[column].quantile(0.10)
+        p90 = df.groupby(altitude_bins)[column].quantile(0.90)
+        p95 = df.groupby(altitude_bins)[column].quantile(0.95)
+        
+        # Calculate the mean of bin_midpoints grouped by altitude bins
+        binmidpoint = df.groupby(altitude_bins)['bin_midpoints'].mean()
+
+        ##Plotting vertprofile starts
+        plot_kwargs_fillbetween = plot_kwargs.copy()
+        del plot_kwargs_fillbetween['marker']
+        del plot_kwargs_fillbetween['markersize']
+        del plot_kwargs_fillbetween['fontsize']
+
+       
+        # Copy the plot_kwargs outside the loop
+        plot_kwargs_fillbox = plot_kwargs.copy()
+        
+
+        # Define box properties for transparency
+        boxprops = {
+            'facecolor': 'none',  # Transparent facecolor
+        }
+
+        # Track labels that have been added to the legend
+        labels_added_to_legend = set()
+        
+        # Plot shaded interquartiles
+        plot_kwargs_fillbetween['label'] = f"{label} (interquartile 25th-75th percentile range)"
+        if interquartile_style == 'shading':
+            ax.fill_betweenx(binmidpoint.values, q1.values, q3.values, alpha=0.3, **plot_kwargs_fillbetween)
```

Comment:
Not necessary right now but in the future it would be nice to have the ability to set the range in the shading method in the control file, potentially with it defaulting to 25th to 75th if nothing else set.